TestFoundation/TestURLResponse.swift

```diff
@@ -31,36 +36,46 @@ class TestURLResponse : XCTestCase {
     
     func test_MIMEType_1() {
         let mimetype = "text/plain"
-        let res = URLResponse(url: URL(string: "test")!, mimeType: mimetype, expectedContentLength: 0, textEncodingName: nil)
+        let res = URLResponse(url: testURL, mimeType: mimetype, expectedContentLength: 0, textEncodingName: nil)
         XCTAssertEqual(res.mimeType, mimetype, "should be the passed in mimetype")
     }
     
     func test_MIMEType_2() {
         let mimetype = "APPlication/wordperFECT"
-        let res = URLResponse(url: URL(string: "test")!, mimeType: mimetype, expectedContentLength: 0, textEncodingName: nil)
+        let res = URLResponse(url: testURL, mimeType: mimetype, expectedContentLength: 0, textEncodingName: nil)
         XCTAssertEqual(res.mimeType, mimetype, "should be the other mimetype")
     }
-
-    func test_ExpectedContentLength() {
-        let zeroContentLength = 0
-        let positiveContentLength = 100
-        let url = URL(string: "test")!
-        let res1 = URLResponse(url: url, mimeType: "text/plain", expectedContentLength: zeroContentLength, textEncodingName: nil)
-        XCTAssertEqual(res1.expectedContentLength, Int64(zeroContentLength), "should be Int65 of the zero length")
-        let res2 = URLResponse(url: url, mimeType: "text/plain", expectedContentLength: positiveContentLength, textEncodingName: nil)
-        XCTAssertEqual(res2.expectedContentLength, Int64(positiveContentLength), "should be Int64 of the positive content length")
+    
+    func test_MIMEType_notAvailable() {
+        let mimetype: String? = nil
+        let res = URLResponse(url: testURL, mimeType: mimetype, expectedContentLength: 0, textEncodingName: nil)
+        XCTAssertEqual(res.mimeType, mimetype, "should be the other mimetype")
+    }
+    
+    func test_ExpectedContentLength_positive() {
+        let contentLength = 100
+        let res1 = URLResponse(url: testURL, mimeType: "text/plain", expectedContentLength: contentLength, textEncodingName: nil)
+        XCTAssertEqual(res1.expectedContentLength, Int64(contentLength), "should be positive Int64 content length")
```

Comment:
Has the test with `zeroContentLength` been removed? In general we don't remove tests unless they are invalid and not required anymore, so I would just add back in the zero content length test to this test function.

Reply:
I took it back in :)
But as a separate one; as I've seen in the other PR you usually prefer less tests in favor of a bit longer ones? Hope its fine, but of course if required could merge them together.